internal/cli/disable.go

```diff
@@ -0,0 +1,84 @@
+package cli
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+type disableCmd struct {
+	all bool
+}
+
+func newDisableCmd() *disableCmd {
+	return &disableCmd{}
+}
+
+func (c *disableCmd) name() model.TiltSubcommand { return "disable" }
+
+func (c *disableCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "disable {-all | <resource>...}",
+		DisableFlagsInUseLine: true,
+		Short:                 "Disables resources",
+		Long: `Disables the specified resources in Tilt.
+
+# disables the resources named 'frontend' and 'backend'
+tilt disable frontend backend
+
+# disables all resources
+tilt disable --all`,
+	}
+
+	cmd.Flags().BoolVar(&c.all, "all", false, "Disable all resources")
+
+	addConnectServerFlags(cmd)
+
+	return cmd
+}
+
+func (c *disableCmd) run(ctx context.Context, args []string) error {
+	ctrlclient, err := newClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if c.all {
+		if len(args) > 0 {
```

Comment:
could we consider combining the if conditions  to reduce the nesting?
```suggestion
	if c.all && len(args) > 0 {
```

Reply:
then the else will trigger when `c.all` is `false` and `len(args) == 0`, which is not what we want